ivc/src/ivc/interpreter.rs

```diff
@@ -36,6 +36,37 @@ use super::{
     LIMB_BITSIZE_XLARGE, N_LIMBS_XLARGE,
 };
 
+pub enum HashSide {
+    Left,
+    Right,
+    RightScaled,
+    Challenges,
+}
+
+/// A trait that defines methods an environment must have to be able to
+/// incrementally verifying the computation.
+pub trait IVCEnv<F: PrimeField, Ff: PrimeField>:
```

Comment:
This trait makes the whole intention of a granular capability access pointless: the code that runs under this environment is fixed forever and cannot be reused for anything else. If everyone's on the same page here than we can proceed. I don't find the design approach convenient; to me it looks like `public static` in java: global state, global effects.

(on top of that we lose composability; but I assume we anyway cannot imagine two `IVCBuilderEnv`s with e.g. different lookup tables, since it's a top-level wrapper)